test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -1620,6 +1620,37 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 				Skip("This cluster does not have a DC-series SKU agent pool")
 			}
 		})
+
+		It("should be able to run an SGX job with sgx-device-plugin", func() {
+			if eng.ExpandedDefinition.Properties.HasDCSeriesSKU() {
+
+				sgx_device_plugin := ""
+				if common.IsKubernetesVersionGe(eng.ExpandedDefinition.Properties.OrchestratorProfile.OrchestratorVersion, "1.17.0") {
+					sgx_device_plugin = "sgx-device-plugin.yaml"
+				} else {
+					sgx_device_plugin = "sgx-device-plugin-before-k8s-1-17.yaml"
+				}
+
+				p, err := pod.CreatePodFromFileIfNotExist(filepath.Join(WorkloadDir, sgx_device_plugin), "sgx-device-plugin", "default", 1*time.Second, cfg.Timeout)
```

Comment:
So, the `CreatePodFromFileIfNotExist` function was designed to `kubectl apply` a `Pod` spec (see https://github.com/Azure/aks-engine/blob/master/test/e2e/kubernetes/workloads/dns-liveness.yaml).

The reason a DaemonSet spec won't work as coded here is that the actual Pod resource rendered by the DaemonSet will have a unique suffix appended to its name. So we can't expect a pod to be present in the cluster literally named "sgx-device-plugin". We *can*, however use that func to kubectl apply the spec (just drop the first response object, e.g.: `_, err := pod.CreatePodFromFileIfNotExist(`...

And then we can use a pattern like this to ensure that all pod resources with a given prefix ("sgx-device-plugin" in this case), are in a Running state:

https://github.com/Azure/aks-engine/blob/master/test/e2e/kubernetes/kubernetes_test.go#L1038

Reply:
Gotcha, I've made the required changes - thank you for your guidance